viewer/testrunner.html

```diff
@@ -1,49 +1,92 @@
+  <link rel="stylesheet" href="node_modules/mocha/mocha.css">
+  <link href="https://maxcdn.bootstrapcdn.com/twitter-bootstrap/2.3.2/css/bootstrap-combined.min.css" rel="stylesheet" integrity="sha384-4FeI0trTH/PCsLWrGCD1mScoFu9Jf2NdknFdFoJhXZFwsvzZ3Bo5sAh7+zL8Xgnd"
+    crossorigin="anonymous">
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/3.2.1/css/font-awesome.css" integrity="sha256-bNPGnNxsIr8mZ4p5VH3uYQorlucOUehl8ml0jm1LZ2I="
+    crossorigin="anonymous" />
+</head>
+
+<body>
+  <div id="test-add-heading-anchors">
+    <h1 id='h1'>Heading 1</h1>
+    <h2 id='h2'>Heading 2</h2>
+    <h3 id='h3'>Heading 3</h3>
+    <h4 id='h4'>Heading 4</h4>
+    <h5 id='h5'>Heading 5</h5>
+    <h6 id='h6'>Heading 6</h6>
+
+    <h1>No-id Heading 1</h1>
+    <h2>No-id Heading 2</h2>
+    <h3>No-id Heading 3</h3>
+    <h4>No-id Heading 4</h4>
+    <h5>No-id Heading 5</h5>
+    <h6>No-id Heading 6</h6>
+    <div class="popovers"></div>
+  </div>
+  <div id="test-popovers">
+      <h1 id='h1'>Heading 1</h1>
+      <h2 id='h2'>Heading 2</h2>
+      <h3 id='h3'>Heading 3</h3>
+      <h4 id='h4'>Heading 4</h4>
+      <h5 id='h5'>Heading 5</h5>
+      <h6 id='h6'>Heading 6</h6>
+
+      <h1>No-id Heading 1</h1>
+      <h2>No-id Heading 2</h2>
+      <h3>No-id Heading 3</h3>
+      <h4>No-id Heading 4</h4>
+      <h5>No-id Heading 5</h5>
+      <h6>No-id Heading 6</h6>
+      <div class="popovers"></div>
+  </div>
+  <div id="test-popovers-timing">
+      <h1 id='h1'>Heading 1</h1>
+      <h2 id='h2'>Heading 2</h2>
+      <h3 id='h3'>Heading 3</h3>
+      <h4 id='h4'>Heading 4</h4>
+      <h5 id='h5'>Heading 5</h5>
+      <h6 id='h6'>Heading 6</h6>
+
+      <h1>No-id Heading 1</h1>
+      <h2>No-id Heading 2</h2>
+      <h3>No-id Heading 3</h3>
+      <h4>No-id Heading 4</h4>
+      <h5>No-id Heading 5</h5>
+      <h6>No-id Heading 6</h6>
+      <div class="popovers"></div>
+  </div>
+
+  <div id="mocha"></div>
+
+  <!--testing libs-->
+  <script src="node_modules/mocha/mocha.js"></script>
+  <script src="node_modules/chai/chai.js"></script>
+  <script src="node_modules/chai-string/chai-string.js"></script>
+  <script src="node_modules/sinon-chai/lib/sinon-chai.js"></script>
+  <script src="node_modules/sinon/pkg/sinon.js"></script>
+  <script>mocha.setup('bdd')</script>
+
+  <!--other dependencies-->
+  <script src="https://code.jquery.com/jquery-1.12.4.js" integrity="sha256-Qw82+bXyGq6MydymqBxNPYTaUXXq7c8v3CwiYwLLNXU=" crossorigin="anonymous"></script>
+  <script src="https://maxcdn.bootstrapcdn.com/twitter-bootstrap/2.3.2/js/bootstrap.min.js" integrity="sha384-vOWIrgFbxIPzY09VArRHMsxned7WiY6hzIPtAIIeTFuii9y3Cr6HE6fcHXy5CFhc"
+    crossorigin="anonymous"></script>
+  <script src="node_modules/clipboard/dist/clipboard.min.js"></script>
+
+  <!--the plugin being tested-->
+  <script src='add-heading-anchors.js'></script>
+
+  <!--the actual test suite-->
+  <script src='test/test-popovers-timing.js'></script>
```

Comment:
the order of tests seems to matter, possibly because this test suite uses fake timers and the other tests don't care about timing at all.  I didn't want to fuss too much about it since we can ensure test ordering right now.